src/Compilers/Core/CommandLine/BuildClient.cs

```diff
@@ -93,13 +93,14 @@ internal int RunCompilation(IEnumerable<string> originalArguments, BuildPaths bu
                         return HandleResponse(buildResponse, parsedArgs, buildPaths);
                     }
                 }
-                catch
+                catch (OperationCanceledException)
```

Comment:
Yes, all server exceptions should be caught by RunServerCompilation. By the time we get here we shouldn't be swallowing exceptions.